pyrefinebio/dataset.py

```diff
@@ -0,0 +1,151 @@
+        self.svd_algorithm = svd_algorithm
+        self.download_url = download_url
+
+    @classmethod
+    def create(
+        cls,
+        data,
+        aggregate_by=None,
+        scale_by=None,
+        email_address=None,
+        email_ccdl_ok=None,
+        start=None,
+        quantile_normalize=None,
+        quant_sf_only=None,
+        svd_algorithm=None
+    ):
+        body = {}
+        body["data"] = data
+        if aggregate_by:
+            body["aggregate_by"] = aggregate_by
+        if scale_by:
+            body["scale_by"] = scale_by
+        if email_address:
+            body["email_address"] = email_address
+        if email_ccdl_ok:
+            body["email_ccdl_ok"] = email_ccdl_ok
+        if start:
+            body["start"] = start
+        if quantile_normalize:
+            body["quantile_normalize"] = quantile_normalize
+        if quant_sf_only:
+            body["quant_sf_only"] = quant_sf_only
+        if svd_algorithm:
+            body["svd_algorithm"] = svd_algorithm
+            
+        response = post_by_endpoint("dataset", payload=body)
+        return Dataset(**response)
+
+    @classmethod
+    def get(cls, id):
+        response = get_by_endpoint("dataset/" + id)
+        return Dataset(**response)
+
+    def update(
+        self,
+        data,
+        aggregate_by=None,
+        scale_by=None,
+        email_address=None,
+        email_ccdl_ok=None,
+        start=None,
+        quantile_normalize=None,
+        quant_sf_only=None,
+        svd_algorithm=None
+    ):
+        body = {}
+        body["data"] = data
+        if aggregate_by:
+            body["aggregate_by"] = aggregate_by
+        if scale_by:
+            body["scale_by"] = scale_by
+        if email_address:
+            body["email_address"] = email_address
+        if email_ccdl_ok:
+            body["email_ccdl_ok"] = email_ccdl_ok
+        if start:
+            body["start"] = start
+        if quantile_normalize:
+            body["quantile_normalize"] = quantile_normalize
+        if quant_sf_only:
+            body["quant_sf_only"] = quant_sf_only
+        if svd_algorithm:
+            body["svd_algorithm"] = svd_algorithm
+
+        response = put_by_endpoint("dataset/" + self.id, payload=body)
+        return Dataset(**response)
+
+    def process(self, email_address):
+        response = self.update(self.data, start=True, email_address=email_address)
+        self.is_processing = response.is_processing
+
+    def check(self):
+        response = self.get(self.id)
+        self.is_processing = response.is_processing
+        self.is_processed = response.is_processed
+        return response.is_processed
+
+    def download(self, path):
+        download_url = self.download_url or self.get(self.id).download_url
+
+        if not download_url:
+            raise DownloadError()
+
+        response = download(download_url)
+
+        with open(path, "wb") as f:
+            f.write(response.content)
```

Comment:
I believe the way this is currently will result in the entire dataset being stored in memory. These datasets could be large, so we probably should use [this answer](https://stackoverflow.com/a/39217788/6095378) to stream the data to disk.